src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -1017,6 +1029,20 @@ public SIPDialog getDialog(String dialogId) {
 
     }
 
+    private SIPDialog findSIPDialogFromInjectionListeners(String dialogId) {
+        SIPDialog dialog = null;
+        synchronized (dialogInjectionListeners) {
```

Comment:
what kind of concurrent access are you trying to sync? In this case, are you trying to prevent multiple invocations to findSIPDialogFromInjectionListeners to be concurrent?if yes, why, it seems to be readOnly scenario?

If you are trying to sync with add/remove writing scenarios, so the iterator is protected against concurrent modifcation, shouldnt you use synchronize on those scenarios as well?

Is it becuase the subsequent invocation to "putDialog"? if yes, why? dialogTable seems to be already synced with a concurrent Map.